RecoBTag/DeepFlavour/interface/TrackInfoBuilder.h

```diff
@@ -0,0 +1,148 @@
+#ifndef RecoBTag_DeepFlavour_TrackInfoBuilder_h
+#define RecoBTag_DeepFlavour_TrackInfoBuilder_h
+
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/Candidate/interface/VertexCompositePtrCandidate.h"
+
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "RecoVertex/VertexTools/interface/VertexDistance3D.h"
+#include "TVector3.h"
+
+namespace btagbtvdeep{
+
+// adapted from DeepNtuples
+class TrackInfoBuilder{
+public:
+    TrackInfoBuilder(edm::ESHandle<TransientTrackBuilder> & build):
+        builder_(build),
+        trackMomentum_(0),
+        trackEta_(0),
+        trackEtaRel_(0),
+        trackPtRel_(0),
+        trackPPar_(0),
+        trackDeltaR_(0),
+        trackPtRatio_(0),
+        trackPParRatio_(0),
+        trackSip2dVal_(0),
+        trackSip2dSig_(0),
+        trackSip3dVal_(0),
+        trackSip3dSig_(0),
+
+        trackJetDistVal_(0),
+        trackJetDistSig_(0)
+{
+
+
+}
+
+    void buildTrackInfo(const reco::Candidate * candidate ,const math::XYZVector&  jetDir, GlobalVector refjetdirection, const reco::Vertex & pv){
+			TVector3 jetDir3(jetDir.x(),jetDir.y(),jetDir.z());
+
+        // deal with PAT/AOD polymorphism to get track
+        const reco::Track * track_ptr = nullptr;
+        auto packed_candidate = dynamic_cast<const pat::PackedCandidate *>(candidate);
+        auto pf_candidate = dynamic_cast<const reco::PFCandidate *>(candidate);
+        if (pf_candidate) {
+          track_ptr = pf_candidate->bestTrack(); // trackRef was sometimes null
+        } else if (packed_candidate) {
+          track_ptr = &(packed_candidate->pseudoTrack());
+        }
+				if(!track_ptr) {
+					TVector3 trackMom3(
```

Comment:
some more consistent indentation would be nice

Reply:
will fix

Reply:
fixed with https://github.com/cms-sw/cmssw/pull/19893/commits/d4a51ecf6c8f631b5dd9b3c6736f4e7ec2c23631